controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/artifact/Artifact.java

```diff
@@ -0,0 +1,95 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.controller.api.integration.artifact;
+
+import com.yahoo.component.Version;
+
+import java.time.Instant;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * A registry artifact (e.g. container image or RPM)
+ *
+ * @author mpolden
+ */
+public class Artifact {
+
+    private final String id;
+    private final Optional<String> registry;
+    private final Optional<String> repository;
+    private final Optional<String> tag;
```

Comment:
Won't there always be some kind of tag/version? This should be non-optional.

Reply:
My point was that an artifact (be it a container image or RPM) will always have some sort of "tag", e.g. a version number, so it can be non-optional.

Also, maybe we should remove the separate `version` field  and parse it from `tag` in the constructor as those should never come from two separate sources.